src/macro-core/macro-condition-tempvar.cpp

```diff
@@ -0,0 +1,368 @@
+	case MacroConditionTempVar::Condition::LESS_THAN_VARIABLE:
+		return validNumbers && val1 < val2;
+	case MacroConditionTempVar::Condition::GREATER_THAN_VARIABLE:
+		return validNumbers && val1 > val2;
+	default:
+		blog(LOG_WARNING,
+		     "Unexpected call of %s with condition type %d", __func__,
+		     static_cast<int>(_type));
+	}
+
+	return false;
+}
+
+bool MacroConditionTempVar::CheckCondition()
+{
+	auto var = _tempVar.GetTempVariable(GetMacro());
+	if (!var) {
+		return false;
+	}
+
+	switch (_type) {
+	case MacroConditionTempVar::Condition::EQUALS:
+		return Compare(*var);
+	case MacroConditionTempVar::Condition::IS_EMPTY:
+		return var->Value().has_value() && var->Value()->empty();
+	case MacroConditionTempVar::Condition::IS_NUMBER:
+		return isNumber(*var);
+	case MacroConditionTempVar::Condition::LESS_THAN:
+		return compareNumber(*var, _numValue, true);
+	case MacroConditionTempVar::Condition::GREATER_THAN:
+		return compareNumber(*var, _numValue, false);
+	case MacroConditionTempVar::Condition::VALUE_CHANGED:
+		return ValueChanged(*var);
+	case MacroConditionTempVar::Condition::EQUALS_VARIABLE:
+		return CompareVariables();
+	case MacroConditionTempVar::Condition::LESS_THAN_VARIABLE:
+		return CompareVariables();
+	case MacroConditionTempVar::Condition::GREATER_THAN_VARIABLE:
+		return CompareVariables();
+	}
+
+	return false;
+}
+
+bool MacroConditionTempVar::Save(obs_data_t *obj) const
+{
+	MacroCondition::Save(obj);
+	_tempVar.Save(obj);
+	obs_data_set_string(obj, "variableName",
+			    GetWeakVariableName(_variable2).c_str());
+	_strValue.Save(obj, "strValue");
+	obs_data_set_double(obj, "numValue", _numValue);
+	obs_data_set_int(obj, "condition", static_cast<int>(_type));
+	_regex.Save(obj);
+	return true;
+}
+
+bool MacroConditionTempVar::Load(obs_data_t *obj)
+{
+	MacroCondition::Load(obj);
+	_tempVar.Load(obj, GetMacro());
+	_variable2 =
+		GetWeakVariableByName(obs_data_get_string(obj, "variableName"));
+	_strValue.Load(obj, "strValue");
+	_numValue = obs_data_get_double(obj, "numValue");
+	_type = static_cast<Condition>(obs_data_get_int(obj, "condition"));
+	_regex.Load(obj);
+	// TODO: remove in future version
+	if (obs_data_has_user_value(obj, "regex")) {
+		_regex.CreateBackwardsCompatibleRegex(
+			obs_data_get_bool(obj, "regex"));
+	}
+	return true;
+}
+
+std::string MacroConditionTempVar::GetShortDesc() const
+{
+	auto var = _tempVar.GetTempVariable(GetMacro());
+	if (!var) {
+		return "";
+	}
+	return var->Name();
+}
+
+static inline void populateConditionSelection(QComboBox *list)
+{
+	for (auto entry : conditionTypes) {
+		list->addItem(obs_module_text(entry.second.c_str()));
+	}
+}
+
+MacroConditionTempVarEdit::MacroConditionTempVarEdit(
+	QWidget *parent, std::shared_ptr<MacroConditionTempVar> entryData)
+	: QWidget(parent),
+	  _variables(new TempVariableSelection(this)),
+	  _variables2(new VariableSelection(this)),
+	  _conditions(new QComboBox()),
+	  _strValue(new VariableTextEdit(this, 5, 1, 1)),
```

Comment:
Since you changed some values to `VariableTextEdit` in another commit and also made it for temp vars, would be nice to also change the same `_strValue` field in Variables condition for both consistency and to ofc enable comparisons like `${1stVar} - ${2ndVar}` also there.

Reply:
I have changed `_strValue` to `VariableTextEdit` with https://github.com/WarmUpTill/SceneSwitcher/pull/901/commits/071a826db784daeda9745f78c9cd48083a41189c already.
I am probably just misunderstanding something.

Reply:
Ah, sorry, too many commits, I probably just missed it